File: app/src/main/java/com/example/architectureapplication/NoteRepository.java
package com.example.architectureapplication;

import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

import javax.inject.Inject;

public class NoteRepository {
    private NoteDao noteDao;
    private LiveData<List<Note>> allNotes;

    @Inject
    public NoteRepository(NoteDao dao) {
        noteDao = dao;
        allNotes = noteDao.getNotes();
    }

    public void insert(Note note){
        new InsertNoteAsyncTask(noteDao).execute(note);
    }

    public void update(Note note){
        new UpdateNoteAsyncTask(noteDao).execute(note);
    }

    public void delete(Note note) {
        new DeleteNoteAsyncTask(noteDao).execute(note);
    }

    public void deleteAllNotes() {
        new DeleteAllNotesAsyncTask(noteDao).execute();
    }

    public LiveData<List<Note>> getAllNotes() {
        return allNotes;
    }

    private static class InsertNoteAsyncTask extends AsyncTask<Note,Void,Void>{
        private NoteDao noteDao;
        public InsertNoteAsyncTask(NoteDao nDao) {
            this.noteDao = nDao;
        }

        @Override
        protected Void doInBackground(Note... notes) {
            noteDao.insert(notes[0]);
            return null;
        }
    }

    private static class UpdateNoteAsyncTask extends AsyncTask<Note,Void,Void>{
        private NoteDao noteDao;
        public UpdateNoteAsyncTask(NoteDao nDao) {
            this.noteDao = nDao;
        }

        @Override
        protected Void doInBackground(Note... notes) {
            noteDao.update(notes[0]);
            return null;
        }
    }

    private static class DeleteNoteAsyncTask extends AsyncTask<Note,Void,Void>{
        private NoteDao noteDao;
        public DeleteNoteAsyncTask(NoteDao nDao) {
            this.noteDao = nDao;
        }

        @Override
        protected Void doInBackground(Note... notes) {
            noteDao.delete(notes[0]);
            return null;
        }
    }

    private static class DeleteAllNotesAsyncTask extends AsyncTask<Void,Void,Void>{
        private NoteDao noteDao;
        public DeleteAllNotesAsyncTask(NoteDao nDao) {
            this.noteDao = nDao;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            noteDao.deleteAllNotes();
            return null;
        }
    }


}

File: app/src/main/java/com/example/architectureapplication/NoteRepository.kt
package com.example.architectureapplication

import android.os.AsyncTask
import androidx.lifecycle.LiveData
import javax.inject.Inject

class NoteRepository @Inject constructor(val dao: NoteDaoKt) {

    val allNotes: LiveData<List<NoteKt>> = dao.getNotes()

    fun insert(note: NoteKt) = InsertNoteAsyncTask(dao).execute(note)

    fun update(note: NoteKt) = UpdateNoteAsyncTask(dao).execute(note)

    fun delete(note: NoteKt) = DeleteNoteAsyncTask(dao).execute(note)

    fun deleteAllNotes() = DeleteAllNoteAsyncTask(dao).execute()

    inner class InsertNoteAsyncTask(val dao: NoteDaoKt) : AsyncTask<NoteKt, Void, Void>() {

        override fun doInBackground(vararg note: NoteKt?): Void? {
            dao.insert(note[0]!!)
            return null
        }

    }

    inner class UpdateNoteAsyncTask(val dao: NoteDaoKt) : AsyncTask<NoteKt, Void, Void>() {

        override fun doInBackground(vararg note: NoteKt?): Void? {
            dao.update(note[0]!!)
            return null
        }

    }

    inner class DeleteNoteAsyncTask(val dao: NoteDaoKt) : AsyncTask<NoteKt, Void, Void>() {

        override fun doInBackground(vararg note: NoteKt?): Void? {
            dao.delete(note[0]!!)
            return null
        }

    }

    inner class DeleteAllNoteAsyncTask(val dao: NoteDaoKt) : AsyncTask<Void, Void, Void>() {

        override fun doInBackground(vararg note: Void?): Void? {
            dao.deleteAllNotes()
            return null
        }

    }
}
